app/src/pages/tasks/task.spec.ts

```diff
@@ -23,15 +23,25 @@ describe('Pages: Tasks', () => {
 
   it('should filter tasks when select a tab', async(() => {
     tabName = "TASKS.PREVIOUS";
-
     instance.showTasks(tabName);
+    fixture.detectChanges();
+    let element = fixture.nativeElement;
+    let nodeList = element.querySelectorAll('.from-date');
+    let fromDates = Array.prototype.slice.call(nodeList);
 
-    expect(instance.list).toEqual(instance.pastTasks);
+    fromDates.forEach(function(date){
+      expect(Date.parse(date.textContent)).toBeLessThan(Date.parse(new Date()));
```

Comment:
it shouldn't test the correct date (that is a task for the server), you may need another fixture with dates for test that the task are rendered